gems/aws-sdk-core/lib/aws-sdk-core/profile/profile_file_standardizer.rb

```diff
@@ -0,0 +1,113 @@
+# frozen_string_literal: true
+
+module AWS::SDK::Core
+  # @api private
+  class ProfileFileStandardizer
+    def initialize(raw_file, file_type)
+      @raw_file = raw_file
+      @file_type = file_type
+    end
+
+    def standardize
+      standardize_profile_file
+      @standardized_file
+    end
+
+    private
+
+    # rubocop:disable Metrics
+    def standardize_profile_file
+      @standardized_file = {}
+
+      @raw_file.each do |raw_profile_name, profile_properties|
+        raw_profile_name_has_profile_prefix =
+          raw_profile_name.start_with?('profile ') ||
+          raw_profile_name.start_with?("profile\t")
+        raw_profile_name_has_sso_session_prefix =
+          raw_profile_name.start_with?('sso-session ') ||
+          raw_profile_name.start_with?("sso-session\t")
+
+        if @file_type == :config
+          if raw_profile_name_has_profile_prefix
+            standardized_profile_name =
+              raw_profile_name[('profile'.length)..].strip
+          elsif raw_profile_name_has_sso_session_prefix
+            standardized_profile_name =
```

Comment:
What happens if there is a profile and an sso-session with the same name?  It seems like stripping the sso-session from it (which I think is what is being done below)? could lead to conflicts.

Reply:
I don't think they can be but I could be wrong